crates/uv/src/commands/tool/run.rs

```diff
@@ -586,25 +619,14 @@ async fn get_or_create_environment(
                 });
         if let Some(environment) = existing_environment {
             // Check if the installed packages meet the requirements.
-            let site_packages = SitePackages::from_environment(&environment)?;
-
-            let requirements = requirements
-                .iter()
-                .cloned()
-                .map(UnresolvedRequirementSpecification::from)
-                .collect::<Vec<_>>();
-            let constraints = [];
-
-            if matches!(
-                site_packages.satisfies(
-                    &requirements,
-                    &constraints,
-                    &interpreter.resolver_marker_environment()
-                ),
-                Ok(SatisfiesResult::Fresh { .. })
-            ) {
-                debug!("Using existing tool `{}`", from.name);
-                return Ok((from, environment));
+            if let Ok(Some(tool_receipt)) = installed_tools.get_tool_receipt(&from.name) {
+                if requirements == tool_receipt.requirements()
+                    && constraints == tool_receipt.constraints()
+                    && overrides == tool_receipt.overrides()
+                {
+                    debug!("Using existing tool `{}`", from.name);
+                    return Ok((from, environment));
+                }
```

Comment:
I don't know if this is a good change. Previously, we just checked if the _environment_ was compatible with the request. Now, we check if the _receipt_ contained the exact same requested dependencies. This is consistent with `uv tool install`, but it's also "stricter".